redhat-cop/auto-docs
# Ansible Automation + API

### What does this PR do?

This is a multi-faceted PR.

Part of the PR is some re-factoring of the structure. All the code for generating the screenshot lies in the `/auto-diagram/generator/` folder. All the code for serving up the image through an API lives in `auto-diagram/API`.

Another part of the PR is the creation of the API functionality. Now, you can generate the image and serve it to the end user through a simple link. This is accomplished through an API endpoint. I used express to set this up.

Another part of the PR is the addition of ansible roles to generate the diagram and kickstart the API



### How should this be tested?
Run `ansible-playbook -i inventory/ playbook.yml` to generate the image and start up the API.

Checkout http://localhost:1234/getIMG to see the PNG.

The README offers many more details on playing around with the diagram.
### Is there a relevant Issue open for this?

Resolves #5 
Resolves #4 
Resolves #3 

### Other Relevant info, PRs, etc.
n/a

### People to notify
cc:
